### PR TITLE
Update TBS docs for 1.2

### PR DESCRIPTION
**Pull request by tylerphelan**
For TBS team to review: this is now all of these:
- https://github.com/pivotal/build-service/issues/792
- https://github.com/pivotal/build-service/issues/798
- https://github.com/pivotal/build-service/issues/777
- https://github.com/pivotal/build-service/issues/799
- https://github.com/pivotal/build-service/issues/766
- https://github.com/pivotal/build-service/issues/797
- https://github.com/pivotal/build-service/issues/782
- https://github.com/pivotal/build-service/issues/803
- https://github.com/pivotal/build-service/issues/801

Which other branches should this be merged with (if any)?

This is only relevant for TAP 1.2

### CODE REVIEW

File: install.md.hbs [hunk 1]
```diff
@@ -273,14 +266,10 @@ this can reuse the `tap-registry` secret created in
...
-> You can find the buildpack and stack artifacts installed with Tanzu Application Platform
-> in the descriptor file on [Tanzu Network](https://network.pivotal.io/products/tbs-dependencies).
-> The current installed version of the descriptor is
-> [100.0.293](https://network.pivotal.io/products/tbs-dependencies#/releases/1086670). Sometimes the dependencies
-> require updates. You can use a
-> [manual process in a CI/CD context](tanzu-build-service/tbs-about.html#dependencies-manual), or
-> an [automatic update process](tanzu-build-service/tbs-about.html#auto-updates)
-> in the background by Tanzu Application Platform.
+> Sometimes the dependencies require updates. You can use a [manual process in a CI/CD context](tanzu-build-service/tbs-about.html#dependencies-manual), or
+> an experimental [automatic update process](tanzu-build-service/tbs-about.html#auto-updates) in the background by Tanzu Application Platform.
```
Comment (matthewmcnew) [inline install.md.hbs:270]: I think instead of mentioning the auto updates here we should probably encourage updating patch releases.

Comment (matthewmcnew) [reply install.md.hbs:270]: As in 1.2 auto updates via tanzunet is supposed to begin being deprecated.

File: multicluster/reference/tap-values-build-sample.md [hunk 2]
```diff
@@ -41,7 +39,6 @@ Where:
...
   * For Google Cloud Registry, use the contents of the service account JSON file.
-- `TANZUNET-USERNAME` and `TANZUNET-PASSWORD` are the email address and password that you use to log in to VMware Tanzu Network. Your VMware Tanzu Network credentials enable you to configure the dependencies updater. This resource accesses and installs the build dependencies (buildpacks and stacks) Tanzu Build Service needs on your cluster. It can also optionally keep these dependencies up to date as new versions are released on VMware Tanzu Network. This credential can also be configured by using a Secret reference. For more information, see [Install Tanzu Build Service](../../tanzu-build-service/install-tbs.html#install-secret-refs).
 - `DESCRIPTOR-NAME` is the name of the descriptor to import. For more information, see [Descriptors](../../tanzu-build-service/tbs-about.html#descriptors). Available options are:
```
Comment (matthewmcnew) [inline multicluster/reference/tap-values-build-sample.md:42]: Can we drop this Descriptor Name?

File: install.md.hbs (hunk 1 above)
Comment (tylerphelan) [reply install.md.hbs]: updated

File: install.md.hbs [hunk 3]
```diff
@@ -273,14 +266,10 @@ this can reuse the `tap-registry` secret created in
...
+> By default, a small-footprint (lite) set of dependencies is installed in TAP which does not contain all Buildpacks and Stacks.
+> To install the larger-footprint (full) set of dependencies which contains all buildpacks and stacks, see [Installing the Full Dependencies](tanzu-build-service/tbs-about.html#auto-updates).
```
Comment (mgibson1121) [inline install.md.hbs]: I think we need to let people know here what additional functionality they get out the full set.  Web server buildpack, airgapped support, PHP.

File: tanzu-build-service/descriptors.md [hunk 4]
```diff
@@ -3,14 +3,11 @@ title: Descriptors
...
-This topic describes the descriptors that are available so you can choose
-which option to configure depending on your use case.
-
 ## <a id="descriptors"></a> About descriptors
 
 Tanzu Build Service descriptors are curated sets of dependencies, including stacks and buildpacks, that are
-continuously released on VMware Tanzu Network to resolve all workload Critical and High CVEs.
-Descriptors are imported into Tanzu Build Service to update the entire cluster.
+continuously released on VMware Tanzu Network to resolve all workload Critical and High CVEs. Descriptors are only used 
+if TAP is configured for automatic dependency updates. Descriptors are imported into Tanzu Build Service to update the entire cluster.
```
Comment (mgibson1121) [inline tanzu-build-service/descriptors.md:10]: Not sure if this is the right place for it.  But we should call out that new buildpacks and stacks are shipped in patch TAP versions.

Comment (tylerphelan) [reply tanzu-build-service/descriptors.md]: good call

File: install.md.hbs (hunk 3 above)
Comment (tylerphelan) [reply install.md.hbs]: let's do it wholesale in https://github.com/pivotal/build-service/issues/766

File: install.md.hbs [hunk 5]
```diff
@@ -273,14 +266,10 @@ this can reuse the `tap-registry` secret created in
...
 > For more information about buildpacks, see the [VMware Tanzu Buildpacks Documentation](https://docs.vmware.com/en/VMware-Tanzu-Buildpacks/index.html).
-> You can find the buildpack and stack artifacts installed with Tanzu Application Platform
-> in the descriptor file on [Tanzu Network](https://network.pivotal.io/products/tbs-dependencies).
-> The current installed version of the descriptor is
-> [100.0.293](https://network.pivotal.io/products/tbs-dependencies#/releases/1086670). Sometimes the dependencies
-> require updates. You can use a
-> [manual process in a CI/CD context](tanzu-build-service/tbs-about.html#dependencies-manual), or
-> an [automatic update process](tanzu-build-service/tbs-about.html#auto-updates)
-> in the background by Tanzu Application Platform.
+> Sometimes the dependencies require updates. You can use a [manual process in a CI/CD context](tanzu-build-service/tbs-about.html#dependencies-manual), or
```
Comment (mgibson1121) [inline install.md.hbs]: I would think that most customers would upgrade dependencies via consuming a new TAP patch following 1.2.  I think that's what we should be highlighting.

File: tanzu-build-service/install-tbs.md [hunk 6]
```diff
@@ -185,10 +152,36 @@ To install the Tanzu Build Service package air-gapped:
...
+tanzu package repository add tbs-full-deps-repository \
+  --url ${INSTALL_REGISTRY_HOSTNAME}/TARGET-REPOSITORY/tbs-full-deps:${VERSION} \
+  --namespace tap-install
+```
+
+5. Install the Full Dependencies package (no `values.yaml` needed):
+
+```console
+tanzu package install full-tbs-deps -p full-tbs-deps.tanzu.vmware.com -v $VERSION -n tap-install
+```
```
Comment (mgibson1121) [inline tanzu-build-service/install-tbs.md:216]: Nice!

File: install.md.hbs [hunk 7]
```diff
@@ -273,14 +266,10 @@ this can reuse the `tap-registry` secret created in
...
+> an experimental [automatic update process](tanzu-build-service/tbs-about.html#auto-updates) in the background by Tanzu Application Platform.
+> By default, a small-footprint (lite) set of dependencies is installed in TAP which does not contain all Buildpacks and Stacks.
```
Comment (mgibson1121) [inline install.md.hbs:270]: Maybe link to this other section you made?  As a reader looking at installing a profile, I wouldn't know which descriptor I should pick based on this description.  ### <a id="descriptor-comparison"></a> Descriptor comparison

File: install.md.hbs [hunk 8]
```diff
@@ -273,14 +266,52 @@ this can reuse the `tap-registry` secret created in
...
+
+```yaml
+buildservice:
+  kp_default_repository: "KP-DEFAULT-REPO"
+  kp_default_repository_username: "KP-DEFAULT-REPO-USERNAME"
+  kp_default_repository_password: "KP-DEFAULT-REPO-PASSWORD"
+  exclude_dependencies: true
+```
+
+2. Get the latest version of the buildservice package:
```
Comment (matthewmcnew) [inline install.md.hbs:290]: Clever!

File: tanzu-build-service/install-tbs.md [hunk 9]
```diff
@@ -231,3 +256,45 @@ Where:
...
+2. Get the latest version of the buildservice package:
+
+```console
+tanzu package available list buildservice.tanzu.vmware.com --namespace tap-install
+```
+
+3. Relocate the TBS full dependencies package repository using the version from the previous step. This should be a similar command to the one used during TAP install:
+
+```console
+imgpkg copy -b registry.tanzu.vmware.com/tanzu-application-platform/full-tbs-deps-package-repo:${VERSION} --to-repo ${INSTALL_REGISTRY_HOSTNAME}/TARGET-REPOSITORY/tbs-full-deps
```
[the remark this answers is not in the document]
Comment (tylerphelan) [reply tanzu-build-service/install-tbs.md:285]: why tar? this is just for full-tbs-dep not offline tbs deps.

File: tanzu-build-service/dependencies.md [hunk 10]
```diff
@@ -0,0 +1,96 @@
...
+- `full`
+- `tiny`
+
+For more information, see [Stacks](https://docs.pivotal.io/tanzu-buildpacks/stacks.html).
+Due to the larger footprint of `full`, installations might take longer.
+The `full` set of dependencies must be installed separately from TAP or TBS which can be found [here](install-tbs.html#tap-install-full-deps).
+
+#### <a id="lite-vs-full-table"></a> Lite vs Full Dependencies Table
+
+Below is a side-by-side comparison of the `lite` and `full` dependencies.
```
Comment (tomkennedy513) [inline tanzu-build-service/dependencies.md:78]: this is great

**Comment by snneji** [comment]
Hi @tylerphelan, is this ready to move out of draft mode so that we can merge it? I'd like to get started on editing it ASAP.

**Comment by tylerphelan** [comment]
@snneji ready!

**Comment by snneji** [comment]
Thanks @tylerphelan

**Pull request merged**